### PR TITLE
feat(reduce): reduce functionality

### PR DESCRIPTION
**Pull request by goanpixie**
Using Reduce to tally votes

### CODE REVIEW

**Comment by goanpixie** [comment]
@kentcdodds Please let me know what you think. Thanks for your feedback and course on egghead.

File: test/reduce-to-tally.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import test from 'ava'
+import reducer from '../src/reduce-to-tally'
+
+test('tallies an array into object with count on each', t => {
+  const initialvalue = {}
+  const original = ["angular", "angular", "rails", "angular", "Python", "angular", "Python", "angular", "Vanilla"]
+  const expected = {"Python": 2, "Vanilla": 1, "angular": 5, "rails": 1}
```
Comment (kentcdodds) [inline test/reduce-to-tally.test.js]: Could you remove the quotes around the keys in this object? They are not necessary :)

File: test/reduce-to-tally.test.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import test from 'ava'
+import reducer from '../src/reduce-to-tally'
+
+test('tallies an array into object with count on each', t => {
+  const initialvalue = {}
+  const original = ["angular", "angular", "rails", "angular", "Python", "angular", "Python", "angular", "Vanilla"]
```
Comment (kentcdodds) [inline test/reduce-to-tally.test.js]: Could you change the double quotes to single quotes?

File: src/reduce-to-tally.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export default reducer
+
+/**
+ * Original Source: https://egghead.io/courses/reduce-data-with-javascript
```
Comment (kentcdodds) [inline src/reduce-to-tally.js]: Normally we're looking for StackOverflow answers in this repo, but I'll let this one slide :)

Comment (goanpixie) [inline src/reduce-to-tally.js]: Sweet. Let me know if the updated changes are good. :)

**Comment by goanpixie** [comment]
Great! Thanks:)

**Pull request merged**